st3/lsp_utils/npm_client_handler.py

```diff
@@ -152,11 +128,13 @@ def on_start(cls, window) -> bool:
             return False
         return cls.__server.ready
 
-    def on_initialized(self, client) -> None:
+    def __init__(self, weaksession: 'weakref.ref[Session]') -> None:
```

Comment:
Another nice thing about this `AbstractPlugin` is that `__init__` is called at the time when the `Session` is initialized, and `__del__` is called when `shutdown` is about to be sent. So, derivatives from `AbstractPlugin` don't have to care about this initialize/initialized/shutdown/exit dance.